src/modules/facility-loan-transaction/facility-loan-transaction.service.ts

```diff
@@ -0,0 +1,50 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs-authentication/acbs-authentication.service';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+
+import { AcbsFacilityLoanTransactionService } from '../acbs/acbs-facility-loan-transaction.service';
+import { GetFacilityLoanTransactionResponseItem } from './dto/get-loan-transaction-response.dto';
+
+@Injectable()
+export class FacilityLoanTransactionService {
+  constructor(
+    private readonly acbsAuthenticationService: AcbsAuthenticationService,
+    private readonly acbsFacilityLoanTransactionService: AcbsFacilityLoanTransactionService,
+    private readonly dateStringTransformations: DateStringTransformations,
+  ) {}
+
+  async getLoanTransactionsByBundleIdentifier(bundleIdentifier: string): Promise<GetFacilityLoanTransactionResponseItem> {
+    const idToken = await this.acbsAuthenticationService.getIdToken();
+    const loanTransaction = await this.acbsFacilityLoanTransactionService.getLoanTransactionByBundleIdentifier(bundleIdentifier, idToken);
+    const loan = loanTransaction.BundleMessageList[0];
+    if (loan.$type !== 'NewLoanRequest') {
+      throw new BadRequestException('Bad request', 'The provided bundleIdentifier does not correspond to a loan transaction.');
+    }
+    const pacAccrual = loan.AccrualScheduleList.find((accrual) => accrual.AccrualCategory.AccrualCategoryCode === 'PAC01');
+    const ctlAccrual = loan.AccrualScheduleList.find((accrual) => accrual.AccrualCategory.AccrualCategoryCode === 'CTL01');
+    const firstAccrual = loan.AccrualScheduleList[0];
+    const firstRepayment = loan.RepaymentScheduleList[0];
+    return {
+      portfolioIdentifier: loanTransaction.PortfolioIdentifier,
+      bundleStatusCode: loanTransaction.BundleStatus.BundleStatusCode,
+      bundleStatusDesc: loanTransaction.BundleStatus.BundleStatusShortDescription,
+      postingDate: this.dateStringTransformations.removeTime(loanTransaction.PostingDate),
+      facilityIdentifier: loan.FacilityIdentifier,
+      borrowerPartyIdentifier: loan.BorrowerPartyIdentifier,
+      productTypeId: loan.ProductType.ProductTypeCode,
+      productTypeGroup: loan.ProductGroup.ProductGroupCode,
+      currency: loan.Currency.CurrencyCode,
+      dealCustomerUsageRate: loan.DealCustomerUsageRate ?? null,
+      dealCustomerUsageOperationType: loan.DealCustomerUsageOperationType?.OperationTypeCode ? loan.DealCustomerUsageOperationType.OperationTypeCode : null,
+      amount: loan.LoanAmount,
+      issueDate: this.dateStringTransformations.removeTime(loan.EffectiveDate),
+      expiryDate: this.dateStringTransformations.removeTime(loan.MaturityDate),
+      spreadRate: pacAccrual.SpreadRate,
+      spreadRateCTL: ctlAccrual.SpreadRate,
```

Comment:
ctlAccrual is not used for BSS facility, can we make this optional?
Issue can be replicated with bundle id 0000257550

Reply:
Good spot, changed this line to `spreadRateCTL: ctlAccrual?.SpreadRate` and added `required: false` to DTO and now returns `200` instead of `500` for bundle id 0000257550. Interestingly I don't see this logic in the Mulesoft code (unless I'm missing something). Are there any other fields that should be optional that I've missed? (e.g., are `pacAccrual`, `firstAccrual` and `firstRepayment` guaranteed to exist?)

Reply:
**If bundle 0000257550 is working then fields should be fine.** 

**More details:**
Historically firstRepayment used to be missing for some facility types, so this is visible in the Mulesoft code, but now all these items will be present: `pacAccrual`, `firstAccrual` and `firstRepayment`.

We split POST Loan transaction to 3 user stories. We will create Loan transaction without child records, in user story 2 we will add Accrual Schedules, and in user story 3 we will add Repayment Schedules. So during development of POST Loan transaction this GET call will not work with these partial Loans, but it is temporary.